torch/csrc/jit/custom_operator.h

```diff
@@ -246,11 +246,24 @@ struct TORCH_API RegisterOperators {
   RegisterOperators& op(
       const std::string& name,
       Implementation&& implementation,
-      OperatorOptions options = OperatorOptions()) {
+      OperatorOptions options) {
+
     registerOperator(createOperator(
         name, std::forward<Implementation>(implementation), options));
     return *this;
   }
+
+  template <typename Implementation>
+  RegisterOperators& op(
+      const std::string& name,
+      Implementation&& implementation) {
+    registrars_.emplace_back(c10::guts::make_unique<c10::RegisterOperators>(name, std::forward<Implementation>(implementation)));
+
+    return *this;
+  }
+
+private:
+  std::vector<std::shared_ptr<c10::RegisterOperators>> registrars_;
```

Comment:
What is going on here? I see a shared_ptr but a make_unique. I am also unclear the reason why the registrars are being kept around. Can a comment be added?

Reply:
`unique_ptr` can be cast to `shared_ptr`, but you're right, `make_shared` would be the better choice here. I changed it. I also added a comment about why we're keeping the registrars around.